src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -130,9 +132,12 @@ public function bulkEditFormAction(Request $request): Response
     public function bulkEditAction(Request $request, int $combinationId): JsonResponse
     {
         try {
+            /** @var CombinationForEditing $combinationForEditing */
+            $combinationForEditing = $this->getQueryBus()->handle(new GetCombinationForEditing($combinationId));
```

Comment:
This query fetches quite a lot of data, besides it will also be executed by the builder so it's too bad it executes twice. It would be nice to use something that is more performant. We could either use the `CombinationRepository::getProductId` or maybe even integrate the productId in the route directly.

It's not that hard to inject in the JS router since `CombinationsManager` already has the productId and could inject it in the `BulkFormHandler`

Reply:
Oh and besides `productId` is already used in the route for the `bulkEditFormAction`